src/pages/user/settings/token.tsx

```diff
@@ -1,14 +1,33 @@
-import { Code, Input, InputGroup, InputRightAddon, Stack, Text } from '@chakra-ui/react';
+import { useGetToken } from '@api';
+import { generateNewApiToken } from '@auth-utils';
+import { Code, Input, InputGroup, InputRightAddon, Stack, Text, useToast } from '@chakra-ui/react';
 import { SettingsLayout, SimpleLink } from '@components';
 import { composeNextGSSP, userGSSP } from '@utils';
 import { GetServerSideProps, GetServerSidePropsContext, InferGetServerSidePropsType } from 'next';
+import { useEffect, useState } from 'react';
 
 const ApiTokenPage = ({}: InferGetServerSidePropsType<typeof getServerSideProps>) => {
-  // TODO get token
-  const token = 'xxxxxxx';
+  const toast = useToast();
+
+  // fetch token
+  const { data } = useGetToken({});
+  const [token, setToken] = useState('');
+
+  useEffect(() => {
+    if (data) {
+      setToken(data.access_token);
+    }
+  }, [data]);
 
   const handleGenerateToken = () => {
-    console.log('generate new token');
+    (async () => {
+      const res = await generateNewApiToken();
```

Comment:
This page will require more than this to keep in sync -- I can work on writing up a handler for this in the useSession hook

Reply:
Ya, let me know what I need to do here.

Reply:
Hi, yeah if you are ready go ahead and merge -- we can work on that as a separate PR. I have a branch off here to start (https://github.com/adsabs/nectar/compare/master...thostetler:nectar:token_settings)
But presumeably we'll need to rethink about it since I've since changed the way we handle the sessions slightly